src/librustc_middle/ty/print/pretty.rs

```diff
@@ -243,6 +258,20 @@ pub trait PrettyPrinter<'tcx>:
         self.try_print_visible_def_path_recur(def_id, &mut callers)
     }
 
+    /// Try to see if this is a unique symbol, meaning there's no other definition with
+    /// the same name.
+    fn try_print_unique_def(mut self, def_id: DefId) -> Result<(Self::Path, bool), Self::Error> {
+        let unique_symbols = &self.tcx().unique_symbols;
+
+        match unique_symbols.get(&def_id) {
+            None => return Ok((self, false)),
+            Some(symbol) => {
+                write!(&mut self, "{}", symbol)?;
```

Comment:
Is this used (as opposed to `write_str`) elsewhere? I forget.

Reply:
Yes, this can be changed to `write_str`. Done.